src/mlpack/methods/softmax_regression/softmax_regression_function.hpp

```diff
@@ -38,12 +37,25 @@ class SoftmaxRegressionFunction
   const arma::mat InitializeWeights();
 
   /**
+   * Initialize Softmax Regression weights(trainable parameters) with
+   * the given parameters.
+   * @param featureSize The features size of the training set
+   * @param numClasses Number of classes for classification.
+   * @param fitIntercept Intercept term flag.
+   * @return weights after initialize
+   */
+  static const arma::mat InitializeWeights(const size_t featureSize,
+                                           const size_t numClasses,
+                                           const bool fitIntercept = false);
```

Comment:
Ah, I can not find any information about rvalue reference being used by Armadillo.

This function do not looks like performance critical part, but the optimization(pass by reference) is quite easy to do.
I like the api of return value more since it do not have any site effect, but pass by reference may get better performance in this case. Maybe provide both of them?